crates/sui-types/src/crypto.rs

```diff
@@ -162,15 +162,22 @@ impl EncodeDecodeBase64 for SuiKeyPair {
         match bytes.first() {
             Some(x) => {
                 if x == &Ed25519SuiSignature::SCHEME.flag() {
-                    let sk = Ed25519PrivateKey::from_bytes(&bytes[1 + Ed25519PublicKey::LENGTH..])?;
+                    let sk = Ed25519PrivateKey::from_bytes(
+                        bytes
```

Comment:
Nit: for these I'd do a separate assignment of `bytes.get(...).ok_or_else(|| ...)?`.